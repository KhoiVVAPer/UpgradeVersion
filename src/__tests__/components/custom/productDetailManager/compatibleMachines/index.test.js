import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CompatibleMachines from '../../../../../components/custom/productDetailManager/compatibleMachines';

const setUp = (props) => {
  const component = shallow(<CompatibleMachines {...props} />);
  return component;
};

describe('components > custom > layoutManager > CompatibleMachines', () => {
  let component;
  const props = {
    layoutData: {
      current: [{ name: 'test' }],
      past: [{ name: 'test' }],
    }
  };
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders heading', () => {
    const input = component.find('SectionHeader').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders proper subheadings', () => {
    const input = component.find({ dcaTest: 'SectionSubHeader' }).length;
    const output = 2;
    expect(input).toEqual(output);
  });

  it('Check if renders two FlatList for current data', () => {
    const input = component.find({ dcaTest: 'currentFlatlist' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if current FlatList has renderItem props', () => {
    const input = component.find({ dcaTest: 'currentFlatlist' }).prop('renderItem');
    expect(input).toBeDefined();
  });

  it('Check if current FlatList has scroll disabled', () => {
    const input = component.find({ dcaTest: 'currentFlatlist' }).prop('scrollEnabled');
    expect(input).toBeFalsy();
  });

  it('Check if current FlatList has data', () => {
    const input = component.find({ dcaTest: 'currentFlatlist' }).prop('data');
    expect(input).toBeDefined();
  });

  it('FlatList past has data numColumns', () => {
    const input = component.find({ dcaTest: 'currentFlatlist' }).prop('numColumns');
    expect(input).toBeDefined();
  });

  it('Check if renders two FlatList for past data', () => {
    const input = component.find({ dcaTest: 'pastFlatlist' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if past FlatList has renderItem props', () => {
    const input = component.find({ dcaTest: 'pastFlatlist' }).prop('renderItem');
    expect(input).toBeDefined();
  });

  it('Check if past FlatList has scroll disabled', () => {
    const input = component.find({ dcaTest: 'pastFlatlist' }).prop('scrollEnabled');
    expect(input).toBeFalsy();
  });

  it('Check if past FlatList has data', () => {
    const input = component.find({ dcaTest: 'pastFlatlist' }).prop('data');
    expect(input).toBeDefined();
  });

  it('FlatList past has data numColumns', () => {
    const input = component.find({ dcaTest: 'pastFlatlist' }).prop('numColumns');
    expect(input).toBeDefined();
  });

  it('Renders navigateToProduct function', () => {
    const componentInstance = component.instance();
    const input = componentInstance.navigateToProduct;
    expect(input).toBeDefined();
  });
});
