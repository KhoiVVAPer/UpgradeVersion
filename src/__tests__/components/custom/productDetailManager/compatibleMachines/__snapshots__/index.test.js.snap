// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > layoutManager > CompatibleMachines Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            COMPATIBLE MACHINES
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            Actual machines
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          
        </react-native-mock>
      </react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            <react-native-mock>
              <react-native-mock>
                test
              </react-native-mock>
            </react-native-mock>
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
    <react-native-mock />
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            Old Machines
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          
        </react-native-mock>
      </react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            <react-native-mock>
              <react-native-mock>
                test
              </react-native-mock>
            </react-native-mock>
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
    <react-native-mock />
  </react-native-mock>
</react-native-mock>
`;
